src/DatabaseConfig.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy;
+
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Str;
+use Stancl\Tenancy\Contracts\Future\CanSetConnection;
+use Stancl\Tenancy\Contracts\ManagesDatabaseUsers;
+use Stancl\Tenancy\Contracts\TenantDatabaseManager;
+use Stancl\Tenancy\Exceptions\DatabaseManagerNotRegisteredException;
+
+class DatabaseConfig
+{
+    /** @var Tenant */
+    public $tenant;
+
+    /** @var callable */
+    public static $usernameGenerator;
+
+    /** @var callable */
+    public static $passwordGenerator;
+
+    /** @var callable */
+    public static $databaseNameGenerator;
+
+    public static function __constructStatic(): void
+    {
+        static::$usernameGenerator = static::$usernameGenerator ?? function (Tenant $tenant) {
+            return Str::random(16);
+        };
+
+        static::$passwordGenerator = static::$usernameGenerator ?? function (Tenant $tenant) {
+            return Hash::make(Str::random(32));
+        };
+
+        static::$databaseNameGenerator = static::$databaseNameGenerator ?? function (Tenant $tenant) {
+            return config('tenancy.database.prefix') . $tenant->id . config('tenancy.database.suffix');
+        };
+    }
+
+    public function __construct(Tenant $tenant)
+    {
+        static::__constructStatic();
+
+        $this->tenant = $tenant;
+    }
+
+    public static function generateDatabaseNameUsing(callable $databaseNameGenerator): void
+    {
+        static::$databaseNameGenerator = $databaseNameGenerator;
+    }
+
+    public static function generateUsernameUsing(callable $usernameGenerator): void
+    {
+        static::$usernameGenerator = $usernameGenerator;
+    }
+
+    public static function generatePasswordUsing(callable $passwordGenerator): void
+    {
+        static::$passwordGenerator = $passwordGenerator;
+    }
```

Comment:
These methods are used to customize the logic used for generating DB names, usernames and passwords.

They should probably be pluralized? E.g. `generateDatabaseNamesUsing()`